src/spatialdata/_core/query/relational_query.py

```diff
@@ -462,14 +464,78 @@ class MatchTypes(Enum):
     no = "no"
 
 
-def join_sdata_spatialelement_table(
-    sdata: SpatialData,
-    spatial_element_name: str | list[str],
-    table_name: str,
-    how: str = "left",
+def _create_sdata_elements_dict_for_join(
+    sdata: SpatialData, spatial_element_name: str | list[str], table_name: str
+) -> tuple[dict[str, dict[str, Any]], AnnData]:
+    assert sdata.tables.get(table_name), f"No table with `{table_name}` exists in the SpatialData object."
```

Comment:
I'll change this to a `ValueError`, see https://github.com/scverse/spatialdata/issues/483